Code/Unsupervised_reconstruction/data/reconstruction_datasets.py

```diff
@@ -1,8 +1,10 @@
+from typing import Generator, Any
 from torch import Tensor
-from torch.utils.data import Dataset, DataLoader
-from data.vocab import computeInferenceData_Samples, computeInferenceData_Cognates, wordsToOneHots
-from source.utils import pad2d_sequence
-from models.types import ModernLanguages, Operations, InferenceData_Samples, InferenceData_Cognates
+from torch.nn.utils.rnn import pad_sequence
+from torch.utils.data import Dataset, IterableDataset, DataLoader, TensorDataset, ChainDataset, get_worker_info
+from data.vocab import computeInferenceData_Samples, computeInferenceData_Cognates, wordsToOneHots, vocabulary
+from Source.utils import pad2d_sequence
+from models.types import ModernLanguages, Operations, InferenceData_Samples, InferenceData_Cognates, PADDING_TOKEN
 
```

Comment:
After a more thorough reading of the _"reconstruction_datasets.py"_ file, it is necessary to review the data flow and structure, in particular to avoid :
- the costs of alternating between object list tuples and object tuple lists
- duplication of data, as found in the file which sends duplicate `dict[ModernLangues, Any]`.

In addition, this file shows the need to implement classes for these objects to make them easier to handle and read.